include/sancus_support/timer.h

```diff
@@ -25,4 +31,21 @@ int  timer_tsc_end(void);
 /* Use for reactive OS support (sancus-support/src/main/main.c) */
 void timer_init(void);
 
+
+#define TIMER_ISR_ENTRY(fct)                                        \
+__attribute__((naked)) __attribute__((interrupt(TIMER_IRQ_VECTOR))) \
+void timerA_isr_entry(void)                                         \
+{                                                                   \
+    __asm__ __volatile__(                                           \
+            "cmp #0x0, r1\n\t"                                      \
+            "jne 1f\n\t"                                            \
```

Comment:
don't you have to save caller-save register R15 etc here before jumping to the ISR (since the ISR function is allowed to overwrite them and the hardware only pushed R1/R2 for an unprotected interrupt)?